src/Components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,42 @@
+import classNames from 'classnames';
+import React from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  return (
+    <div
+      className={classNames(
+        'todo',
+        { completed: todo.completed },
+      )}
+    >
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+        />
+      </label>
+
+      <span className="todo__title">{todo.title}</span>
+
+      {/* Remove button appears only on hover */}
+      <button
+        type="button"
+        className="todo__remove"
+      >
+        ×
+      </button>
+
+      {/* overlay will cover the todo while it is being updated */}
+      <div className="modal overlay">
```

Comment:
Conditionally render the overlay and loader only when the todo is being updated.